src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -175,63 +165,54 @@ else if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user))
     }
 
     /**
-     * GET /exercises/upcoming : Find all exercises that have an upcoming due date.
+     * GET exercises/upcoming : Find all exercises that have an upcoming due date.
      *
      * @return the ResponseEntity with status 200 (OK) and a list of exercises.
      */
-    @GetMapping("/exercises/upcoming")
+    @GetMapping("exercises/upcoming")
     @PreAuthorize("hasRole('ADMIN')")
     public ResponseEntity<Set<Exercise>> getUpcomingExercises() {
         log.debug("REST request to get all upcoming exercises");
-
-        if (!authCheckService.isAdmin()) {
-            return forbidden();
-        }
-
+        authCheckService.checkIsAdminElseThrow(null);
```

Comment:
why do we pass `null` here? the method is supposed to get a `User` object. This way, why would it return anything else than returning an error.
````
    public void checkIsAdminElseThrow(@Nullable User user) {
        if (!isAdmin(user)) {
            throw new AccessForbiddenException(NOT_ALLOWED);
        }
    }